kura/org.eclipse.kura.core/src/main/java/org/eclipse/kura/core/data/DataServiceImpl.java

```diff
@@ -913,7 +914,11 @@ public void run() {
                 }
                 connected = true;
             } catch (KuraConnectException | KuraStoreException e) {
-                logger.warn("Connect failed", e);
+                logger.warn("DataService failure occured: ", e);
+                
+                if (e instanceof KuraStoreException) {
+                    DataServiceImpl.this.watchdogService.checkin(DataServiceImpl.this);
```

Comment:
We should do this only if `DataServiceImpl.this.dataServiceOptions.isConnectionRecoveryEnabled()`

Reply:
Is this equivalent to adding an additional `|| e instanceof KuraStoreException` to line 924?

Reply:
Seems to be so, making that change